pages/[id].js

```diff
@@ -58,6 +53,21 @@ export default function DetailsPage({
       router.push(`/`);
     }
   }
+  const { data: plant, error, isLoading } = useSWR(`/api/plants/${id}`);
+
+  if (error) {
+    return <p>Could not fetch data!</p>;
+  }
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (!plant) {
+    return;
+  }
```

Comment:
In case if `plant` is not found, you page will be blank without any information. Instead if you include return `<p>Plant not found </p>`, then user will have clear information of what just happened.